src/profile/ProfileForm.tsx

```diff
@@ -37,585 +23,315 @@ interface ChangedUser {
-              <Form.Control.Feedback type="invalid">
-                <ul>
-                  {state.errors.username?.map((msg) => (
-                    <li key={msg}>{ERRORS[msg]}</li>
-                  ))}
-                </ul>
-              </Form.Control.Feedback>
-            </>
-          )}
+              },
+            })}
+          />
         </Col>
       </Form.Group>
-
-      <Form.Group as={Row} controlId="profileEmail">
+      <Form.Group as={Row}>
         <Form.Label column sm={2}>
-          Email
+          Name:
         </Form.Label>
-        <Col sm={10}>
-          <Form.Control
-            type="email"
-            name="email"
-            required
-            placeholder="example@example.com"
-            value={state.changedUser.email ?? user?.email ?? ""}
-            onChange={onChange}
-            isValid={(state.changedUser.email?.length ?? 0) > 0}
-            isInvalid={state.errors.email && state.errors.email.length > 0}
+        <Col sm={5}>
+          <Field
+            errors={props.context.errors}
+            type="text"
+            name="first_name"
+            placeholder="First Name"
+            ref={props.context.register({
+              maxLength: {
+                value: 30,
+                message: "USER.FIRST_NAME.TOO_LONG.30",
+              },
+              required: true,
+            })}
+          />
+        </Col>
+        <Col sm={5}>
+          <Field
+            errors={props.context.errors}
+            type="text"
+            name="last_name"
+            placeholder="Last Name"
+            ref={props.context.register({
+              maxLength: {
+                value: 150,
+                message: "USER.LAST_NAME.TOO_LONG.150",
+              },
+              required: true,
+            })}
           />
-          <Form.Control.Feedback type="invalid">
-            <ul>
-              {state.errors.email?.map((msg) => (
-                <li key={msg}>{ERRORS[msg]}</li>
-              ))}
-            </ul>
-          </Form.Control.Feedback>
         </Col>
       </Form.Group>
-
-      <Form.Group as={Row} controlId="profileFirstLastName">
+      <Form.Group as={Row} controlId="email">
         <Form.Label column sm={2}>
-          Name
+          Email:
         </Form.Label>
         <Col sm={10}>
-          <Form.Control
-            name="name"
-            placeholder="e.g. Johnny Appleseed"
-            value={
-              state.changedUser.name ??
-              (user
-                ? `${user.first_name}${
-                    user.last_name ? ` ${user.last_name}` : ""
-                  }`
-                : "")
-            }
-            onChange={onChange}
-            isValid={"first_name" in state.changedUser}
+          <Field
+            errors={props.context.errors}
+            type="email"
+            name="email"
+            ref={props.context.register({
+              required: "USER.REQUIRED.EMAIL",
```

Comment:
`EMAIL` should be lowercased.

Reply:
https://github.com/UWmathNEWS/slugline-web/pull/13/commits/53ece12a397300e594d32c8dc8ce2d7c9859a9ed